### ISSUE TITLE
Remove and cleanup `cleanAppId()`

### ISSUE THREAD

**Issue by LukasReschke** [owncloud/core#9739]
With https://github.com/owncloud/core/pull/1569 we have introduced a function called `cleanAppId()` which removes all potential dangerous characters (e.g. `/` or `\`) from an application id.

This was there to prevent attacks like `core/ajax/translations.php?app=files/../../data/admin/files/` which could have lead to a code execution since ownCloud will include a language file from the specified directory. (e.g. `de_DE.php` inside the user's data folder)

As a quick-fix this solution was ok. But we should get rid of it since opt-in security solutions are always bad. If just a `cleanAppId` gets forgotten we might expose us of different security risks.

Additionally, the usual developer does not really understand when this function should get used and when not. Let's make this situation better.

In short:
- [ ] Remove `cleanAppId()` where it is not needed anymore
- [ ] Move the security checks to the business logic (e.g. https://github.com/owncloud/core/blob/d67d9566ce37e63c4df62d1f57fb03540421895c/settings/ajax/disableapp.php is a bad example)

**Comment by DeepDiver1975** [comment]
@LukasReschke still valid? THX

**Comment by LukasReschke** [comment]
Unfortunately: Yes. - Kinda depends on an refactoring of the OC_Apps stuff.

I think @icewind1991 is working on a refactoring somewhere (?) once that is done I can modify it properly to ensure that we can get rid of that. Before this is a little bit too risky for me to touch. :-)

**Comment by DeepDiver1975** [comment]
> Before this is a little bit too risky for me to touch. :-)

Oh - come one - no risk - no fun!

**Comment by cmonteroluque** [comment]
@Xenopathic can you have a look?

**Comment by Xenopathic** [comment]
@cmonteroluque Do we really want to try and fix technical debt this close to the code freeze?
